### PR TITLE
Revert to default journald rate limit

### PR DESCRIPTION
**Pull request by jongiddy**
## High-level description

Remove the increased journald rate limit. Scale testing demonstrates that the raised limit causes problems for other components that get delayed or lost logs. Logging that exceeds the default indicates that a component is logging excessively, and should be remedied in that component.

## Corresponding DC/OS tickets (required)

  - [DCOS-53763](https://jira.mesosphere.com/browse/DCOS-53763) "too many" Admin Router access log lines went missing in MWST 15

## Related tickets (optional)

  - [DCOS-ID](https://jira.mesosphere.com/browse/DCOS-<number>) JIRA title / short description.

## Checklist for component/package updates:

If you are changing components or packages in DC/OS (e.g. you are bumping the sha or ref of anything underneath `packages`), then in addition to the above please also include:

  - [x] Change log from the last version integrated (this should be a link to commits for easy verification and review): [example](https://github.com/dcos/dcos-mesos-modules/compare/f6fa27d7c40f4207ba3bb2274e2cfe79b62a395a...6660b90fbbf69a15ef46d0184e36755881d6a5ae)
  - [ ] Included a test which will fail if code is reverted but test is not. If there is no test please explain.
Change to logging only.
  - [ ] Test Results: [link to CI job test results for component]
  - [ ] Code Coverage (if available): [link to code coverage report]

### CODE REVIEW

**Comment by jongiddy** [comment]
@mesosphere-mergebot label Ready For Review

**Comment by jgehrcke** [comment]
Thanks for taking on the ownership on this one, @jongiddy!

File: gen/cloud-config.yaml [hunk 1]
```diff
@@ -15,8 +15,6 @@ root:
     content: |
       [Journal]
       MaxLevelConsole=warning
-      RateLimitInterval=1s
-      RateLimitBurst=20000
```
Comment (jgehrcke) [inline gen/cloud-config.yaml]: Do we really want to rely on the default or "set our default explicitly"? :) What is the default? Maybe add a code comment here?

File: CHANGES.md [hunk 2]
```diff
@@ -64,6 +64,8 @@ Please follow the [`CHANGES.md` modification guidelines](https://github.com/dcos
 
 * Prune VIPs with no backends in order to avoid unbounded growth of state and messages exchanged among `dcos-net` processes. (DCOS_OSS-5356)
 
+* DC/OS no longer increases the rate limit for journald logging. (DCOS-53763)
```
Comment (jgehrcke) [inline CHANGES.md]: Can we add one or two more sentences explaining your beautiful findings and reasoning?

**Comment by jgehrcke** [comment]
@jongiddy once again thanks for your efforts on this front. 

I have two questions that are based on curiosity, independent of approving this PR:

1) The motivation for this change is to not lose log important log messages as of journal overload. Do we understand how much this change contributes to this goal? Do we _know_ that when journald limits the log ingress of _one_ component that it actually has more ingress capacity for other components? Even better, do we understand how efficiently the journal can protect itself here?

2) Should we start setting lower rates for specific components of which we know, thanks to your measurements, that they burst much more than other components? The Mesos master could be such a candidate: we wouldn't even lose much: the Mesos master logs to file anyway (properly log-rotated).

File: gen/cloud-config.yaml (hunk 1 above)
Comment (jongiddy) [reply gen/cloud-config.yaml]: I'm wary of adding comments for configuration that we don't override. Like, why do we not have a comment explaining why we're using the journald default for `SyncIntervalSec`?

On the other hand, I'd like to have an explicit comment on why we set `MaxLevelConsole=warning`.

**Comment by jongiddy** [comment]
> Do we understand how much this change contributes to this goal? Do we know that when journald limits the log ingress of one component that it actually has more ingress capacity for other components?

It's circumstantial evidence, but I think I've established that Admin Router drops log messages at the same time that other components are logging excessively. The real test will be the next Scale Test, where we hope to see no instances of Admin Router dropping messages, while also looking at how often the excessive loggers have messages dropped.

> Should we start setting lower rates for specific components of which we know, thanks to your measurements, that they burst much more than other components?

Possibly. I think we should see what we get with this change, and then consider further actions. There are instances in CI where similar problems occur, but the log rate is lower than the default limit, so would not be fixed by this change. Those appear to be due to how disks are allocated in CI. Perhaps more selective log rate management is the answer for those problems.

File: CHANGES.md [hunk 3]
```diff
@@ -64,6 +64,8 @@ Please follow the [`CHANGES.md` modification guidelines](https://github.com/dcos
 
 * Prune VIPs with no backends in order to avoid unbounded growth of state and messages exchanged among `dcos-net` processes. (DCOS_OSS-5356)
 
+* DC/OS no longer increases the rate limit for journald logging.  Scale testing demonstrated that raising the limit overloads journald, causing problems for other components that see delayed or lost logs or, worse, hang until log buffers are read. The default of 10000 messages per 30 seconds appears to distinguish well between busy components and excessively verbose components. (DCOS-53763)
+
```
Comment (jgehrcke) [inline CHANGES.md]: Lovely, thank you!

**Comment by makkes** [comment]
I'm not sure I understand the issue. As far as I understand it, the rate limiting was introduced in August, 2016, wasn't it? Why is this causing problems just now?

File: gen/cloud-config.yaml [hunk 4]
```diff
@@ -15,8 +15,6 @@ root:
     content: |
       [Journal]
       MaxLevelConsole=warning
-      RateLimitInterval=1s
```
Comment (makkes) [inline gen/cloud-config.yaml]: So, in the latest DC/OS version, we're logging so much that we reach the limit of what journald can handle, right? So, we want to fall back to the journald defaults (which probably are tested by the systemd team and journald should be able to cope with).

But isn't the real problem that we are logging way too much? I stumbled the huge amount of log messages even in small clusters as well and it's a pain to rummage through those logs.

**Comment by jgehrcke** [comment]
> Why is this causing problems just now?

It has always caused problems. But we never really discovered them systematically, prioritized working on this. Really this required us to do a couple of MWTs.

**Comment by jgehrcke** [comment]
>  But I'd love to see tickets for tracking the real problem here: That we log way too much.

Example: we lost valuable Admin Router logs. The error message is "send() to syslog failed while logging request". This is tracked by https://jira.mesosphere.com/browse/DCOS-53763 (in description of the PR). 

More context: https://jira.mesosphere.com/browse/DCOS-43956, https://jira.mesosphere.com/browse/DCOS-43592, https://jira.mesosphere.com/browse/DCOS-53208

**Comment by makkes** [comment]
> > But I'd love to see tickets for tracking the real problem here: That we log way too much.
> 
> Example: we lost valuable Admin Router logs. The error message is "send() to syslog failed while logging request". This is tracked by https://jira.mesosphere.com/browse/DCOS-53763 (in description of the PR).
> 
> More context: https://jira.mesosphere.com/browse/DCOS-43956, https://jira.mesosphere.com/browse/DCOS-43592, https://jira.mesosphere.com/browse/DCOS-53208

What I meant is, shouldn't we identify the components that are logging so much and try to decrease the amount of log messages they produce?

**Comment by jongiddy** [comment]
@makkes Before this change, if a component logged too much, then journald slowed down and all components suffered, even those that mostly do the right thing. With this change, only the components that log too much will suffer, by having their log messages dropped.

This puts the pressure on the verbose components to fix their logging. If they don't, or if they have a rare case that exceeds the limit, then at least they won't affect the running cluster. Some issues that are relevant:
https://jira.mesosphere.com/browse/DCOS-42131
https://jira.mesosphere.com/browse/DCOS-56615

**Comment by jongiddy** [comment]
Override please:
```
@mesosphere-mergebot override-status teamcity/dcos/test/terraform/aws/onprem/static/group3 https://jira.mesosphere.com/browse/DCOS-57318
```

**Comment by jgehrcke** [comment]
@mesosphere-mergebot override-status teamcity/dcos/test/terraform/aws/onprem/static/group3 https://jira.mesosphere.com/browse/DCOS-57318

**Comment by jongiddy** [comment]
@mesosphere-mergebot request-review @tillt

**Pull request merged**